test/tests.py

```diff
@@ -0,0 +1,143 @@
+from operations import create_index, check_index, create_document, delete_document, get_document, delete_index
+from operations import check_document
+from http import HTTPStatus
+import unittest
+import os
+import logging
+import time
+import requests
+
+logger = logging.getLogger(__name__)
+
+# Tests will say which line the test failed at and what was the result of the execution, but better logging can be done.
+
+
+def retry_request(request, args=(), max_attempts=10, delay=0.5, expectedStatusCode=None):
+    for attempt in range(1, max_attempts + 1):
+        try:
+            result = request(*args)
+            if result.status_code == expectedStatusCode:
+                return result
+            else:
+                logger.warning(f"Status code returned: {result.status_code} did not"
+                               f" match the expected status code: {expectedStatusCode}."
+                               f" Trying again in {delay} seconds.")
+                time.sleep(delay)
+        except Exception:
+            logger.error(f"All {max_attempts} attempts failed.")
+            raise Exception(f"Couldn't get the expected status code: {expectedStatusCode} even after attempting to"
```

Comment:
I think particularly important to include which request it is in the exception message.